qt-qml/src/tasks/acquire-port.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2025 The Qt Company Ltd.
+// SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only
+
+import * as vscode from 'vscode';
+import getPort from 'get-port';
+
+import { telemetry } from 'qt-lib';
+
+interface AcquirePortTaskDefinition extends vscode.TaskDefinition {
+  /**
+   * The task name
+   */
+  task: string;
+}
+
+export let compoundPort: string | undefined = undefined;
```

Comment:
So this is still a global?

Reply:
Yup, I think there is no advantage of writing to statemanager if it is single workspace. When it comes to multi-workspace, we can only get selected workspace information from the cmake extension which we don't implement it on our side yet.

Reply:
I thought thats what the "WorkspaceStateManager" is for?